File: files/200-299/256/CH-256.R
library(tidyverse)
library(readxl)

path = "files/200-299/256/CH-256 Table Transformation.xlsx"
input = read_excel(path, range = "B2:H6")
test  = read_excel(path, range = "B10:E14")

result = cbind(
    input[1],
    set_names(
        map_dfc(
            c("A", "B", "C"),
            ~ rowSums(input[startsWith(names(input), .x)], na.rm = TRUE)
        ),
        c("A", "B", "C")
    )
)

all.equal(result, test, check.attributes = FALSE)
# > [1] TRUE

File: files/200-299/256/CH-256.py
import pandas as pd

path = "200-299/256/CH-256 Table Transformation.xlsx"
input = pd.read_excel(path, usecols="B:H", nrows=5, skiprows=1)
test = pd.read_excel(path, usecols="B:E", nrows=5, skiprows=9)

result = (input.groupby('Date').sum()
    .T.groupby(lambda x: x[0]).sum()
    .T.reset_index())
for col in result.columns:
     if col != 'Date':
          result[col] = result[col].astype(int)

print(result.equals(test)) # True